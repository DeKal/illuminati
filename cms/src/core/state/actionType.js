export const ERROR = 'app/error'
export const APP_TURN_ON_LOADING = 'app/loading/on'
export const APP_TURN_OFF_LOADING = 'app/loading/off'
export const OPEN_TOAST = 'app/toast/open'
export const CLOSE_TOAST = 'app/toast/close'

export const OPEN_DRAWER = 'drawer/openDrawer'
export const CLOSE_DRAWER = 'drawer/closeDrawer'

export const CHANGE_PAGE = 'posts/changePage'
export const CHANGE_ROW_PER_PAGE = 'posts/changeRowPerPage'
export const SET_ORDER = 'posts/setOrder'
export const SET_ORDER_COLUMN = 'posts/setOrderColumn'
export const SET_SELECTED_POST = 'posts/setSelectedPost'
export const SELECT_ALL_POSTS = 'posts/selectAllPosts'
export const CLEAR_SELECTED = 'posts/clearSelected'
export const FETCH_POSTS = 'posts/fetchPosts'

export const FETCH_POST = 'post/fetchPost'
export const EDIT_POST_TITLE = 'post/editPostTitle'
export const EDIT_POST_SLUG = 'post/editPostSlug'
export const EDIT_POST_BRIEF = 'post/editPostBrief'
export const EDIT_POST_AUTHOR = 'post/editPostAuthor'
export const EDIT_POST_TIME = 'post/editPostTime'
export const EDIT_POST_TAGS = 'post/editPostTags'
export const EDIT_POST_CONTENT = 'post/editPostContent'
export const SAVE_EDITED_CONTENT = 'post/saveEditedContent'
export const CREATE_NEW_POST = 'post/createNewPost'
export const INIT_POST = 'post/initPost'
