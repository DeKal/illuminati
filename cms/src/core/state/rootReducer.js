import { combineReducers } from 'redux'

import drawer from 'core/state/drawer/reducers'
import posts from 'posts/state/reducers'

export default combineReducers({
  drawer,
  posts
})
