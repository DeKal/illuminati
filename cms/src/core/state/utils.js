export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
