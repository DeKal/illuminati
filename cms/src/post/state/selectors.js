export const getPostTitle = state => state.post.content.title

export const getEditedTitle = state => state.post.editedContent.title

export const getPostSlug = state => state.post.content.slug

export const getEditedSlug = state => state.post.editedContent.slug

export const getPostBrief = state => state.post.content.brief

export const getEditedBrief = state => state.post.editedContent.brief

export const getPostAuthor = state => state.post.content.author

export const getEditedAuthor = state => state.post.editedContent.author

export const getPostTime = state => state.post.content.time

export const getEditedTime = state => state.post.editedContent.time

export const getPostTags = state => state.post.content.tag

export const getEditedTags = state => state.post.editedContent.tag

export const getEditedContent = state => state.post.editedContent.content

export const getSavingContent = state => state.post.editedContent

export const getIsPostLoading = state => state.post.loading
