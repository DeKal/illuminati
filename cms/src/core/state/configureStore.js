import { createStore } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import rootReducer from 'core/state/rootReducer'

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  whitelist: ['drawer']
}

const persistedReducer = persistReducer(persistConfig, rootReducer)

const configureStore = initState => {
  const store = createStore(persistedReducer, initState)
  const persistor = persistStore(store)
  return { store, persistor }
}

export default configureStore
