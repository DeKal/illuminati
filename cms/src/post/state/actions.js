import { createAction } from 'redux-actions'
import {
  FETCH_POST,
  EDIT_POST_TITLE,
  EDIT_POST_SLUG,
  EDIT_POST_BRIEF,
  EDIT_POST_AUTHOR,
  EDIT_POST_TIME,
  EDIT_POST_TAGS,
  EDIT_POST_CONTENT,
  SAVE_EDITED_CONTENT
} from 'core/state/actionType'
import { toRequest } from 'core/state/utils'

export const fetchPost = createAction(toRequest(FETCH_POST))
export const editPostTitle = createAction(EDIT_POST_TITLE)
export const editPostSlug = createAction(EDIT_POST_SLUG)
export const editPostBrief = createAction(EDIT_POST_BRIEF)
export const editPostAuthor = createAction(EDIT_POST_AUTHOR)
export const editPostTime = createAction(EDIT_POST_TIME)
export const editPostTags = createAction(EDIT_POST_TAGS)
export const editPostContent = createAction(EDIT_POST_CONTENT)
export const saveEditedContent = createAction(toRequest(SAVE_EDITED_CONTENT))
