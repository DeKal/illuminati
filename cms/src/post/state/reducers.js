import {
  FETCH_POST,
  EDIT_POST_TITLE,
  EDIT_POST_SLUG,
  EDIT_POST_BRIEF,
  EDIT_POST_AUTHOR,
  EDIT_POST_TIME,
  EDIT_POST_TAGS,
  EDIT_POST_CONTENT,
  SAVE_EDITED_CONTENT,
  CREATE_NEW_POST,
  INIT_POST
} from 'core/state/actionType'
import { toSuccess, toError, toRequest } from 'core/state/utils'
import { arrayToAttributes } from 'core/utils/misc'

import format from 'date-fns/format'

const post = (state = {}, action) => {
  switch (action.type) {
    case toRequest(FETCH_POST): {
      return {
        ...state,
        loading: true
      }
    }
    case toSuccess(FETCH_POST): {
      return {
        ...state,
        loading: false,
        content: {
          ...action.payload,
          tag: arrayToAttributes(action.payload.tag)
        },
        editedContent: {
          ...action.payload,
          tag: arrayToAttributes(action.payload.tag)
        }
      }
    }
    case toError(FETCH_POST): {
      throw action.error
    }
    case EDIT_POST_TITLE: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          title: action.payload
        }
      }
    }
    case EDIT_POST_SLUG: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          slug: action.payload
        }
      }
    }
    case EDIT_POST_BRIEF: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          brief: action.payload
        }
      }
    }
    case EDIT_POST_AUTHOR: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          author: action.payload
        }
      }
    }
    case EDIT_POST_TIME: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          time: format(action.payload, 'dd/MM/yyyy')
        }
      }
    }
    case EDIT_POST_TAGS: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          tag: action.payload
        }
      }
    }
    case EDIT_POST_CONTENT: {
      return {
        ...state,
        editedContent: {
          ...state.editedContent,
          content: action.payload
        }
      }
    }
    case toSuccess(SAVE_EDITED_CONTENT): {
      return {
        ...state,
        content: {
          ...state.editedContent
        }
      }
    }
    case toError(SAVE_EDITED_CONTENT): {
      return state
    }
    case toSuccess(CREATE_NEW_POST): {
      return {
        ...state,
        content: {
          ...state.editedContent
        }
      }
    }
    case toError(CREATE_NEW_POST): {
      return state
    }
    case INIT_POST: {
      return {
        ...state,
        content: {
          id: '',
          slug: '',
          title: '',
          brief: '',
          time: Date.now(),
          tag: {},
          author: '',
          content: '',
          viewNum: 0,
          commentNum: 0,
          likeNum: 0
        },
        editedContent: {
          id: '',
          slug: '',
          title: '',
          brief: '',
          time: Date.now(),
          tag: {},
          author: '',
          content: '',
          viewNum: 0,
          commentNum: 0,
          likeNum: 0
        }
      }
    }
    default:
      return state
  }
}

export default post
