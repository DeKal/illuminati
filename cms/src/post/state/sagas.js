import { call, put, takeLatest, select } from 'redux-saga/effects'
import { fetchPostFromSlug, savingPost } from 'post/api/fetch'
import { FETCH_POST, SAVE_EDITED_CONTENT } from 'core/state/actionType'
import { toSuccess, toError, toRequest } from 'core/state/utils'
import { getSavingContent } from 'post/state/selectors'

function* fetchPost({ payload: slug }) {
  try {
    const post = yield call(fetchPostFromSlug, slug)
    yield put({ type: toSuccess(FETCH_POST), payload: post })
  } catch (e) {
    yield put({ type: toError(FETCH_POST), message: e.message })
  }
}

function* savePost() {
  const state = yield select()
  try {
    yield call(savingPost, getSavingContent(state))
    yield put({ type: toSuccess(SAVE_EDITED_CONTENT), payload: '' })
  } catch (e) {
    yield put({ type: toError(SAVE_EDITED_CONTENT), message: e.message })
  }
}

function* postSaga() {
  yield takeLatest(toRequest(FETCH_POST), fetchPost)
  yield takeLatest(toRequest(SAVE_EDITED_CONTENT), savePost)
}

export default postSaga
