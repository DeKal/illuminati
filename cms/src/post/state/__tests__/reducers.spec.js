import postReducer from '../reducers'
import {
  FETCH_POST,
  EDIT_POST_TITLE,
  EDIT_POST_SLUG,
  EDIT_POST_BRIEF,
  EDIT_POST_AUTHOR,
  EDIT_POST_TIME,
  EDIT_POST_TAGS,
  EDIT_POST_CONTENT,
  SAVE_EDITED_CONTENT,
  CREATE_NEW_POST,
  INIT_POST
} from 'core/state/actionType'
import { toRequest, toSuccess, toError } from 'core/state/utils'

describe('Post Reducer', () => {
  describe('default', () => {
    it('should work corectly', function() {
      const action = {
        type: 'default'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
      expect(() => {
        postReducer()
      }).toThrow(TypeError)
    })
  })

  describe('FETCH_POST', () => {
    it('toRequest should work corectly', function() {
      const action = {
        type: toRequest(FETCH_POST)
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })

    it('toSuccess should work corectly', function() {
      const action = {
        type: toSuccess(FETCH_POST),
        payload: {
          id: 'test',
          slug: 'test',
          title: 'test',
          brief: 'test',
          time: 'test',
          tag: ['test', 'test1'],
          author: 'test',
          content: 'test'
        }
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })

    it('toError should work corectly', function() {
      const action = {
        type: toError(FETCH_POST),
        error: new Error('test')
      }
      expect(() => {
        postReducer({}, action)
      }).toThrow(Error)
    })
  })

  describe('CREATE_NEW_POST', () => {
    it('toSuccess should work corectly', function() {
      const state = {
        editedContent: {
          id: 'test',
          slug: 'test',
          title: 'test',
          brief: 'test',
          time: 'test',
          tag: ['test', 'test1'],
          author: 'test',
          content: 'test'
        }
      }
      const action = {
        type: toSuccess(CREATE_NEW_POST),
        payload: {}
      }
      expect(postReducer(state, action)).toMatchSnapshot()
    })
  })

  describe('SAVE_EDITED_CONTENT', () => {
    it('toSuccess should work corectly', function() {
      const state = {
        editedContent: {
          id: 'test',
          slug: 'test',
          title: 'test',
          brief: 'test',
          time: 'test',
          tag: ['test', 'test1'],
          author: 'test',
          content: 'test'
        }
      }
      const action = {
        type: toSuccess(SAVE_EDITED_CONTENT),
        payload: {}
      }
      expect(postReducer(state, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_TITLE', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_TITLE,
        payload: 'edited title'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_SLUG', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_SLUG,
        payload: 'edited-slug'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_BRIEF', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_BRIEF,
        payload: 'edited brief'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_AUTHOR', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_AUTHOR,
        payload: 'edited author'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_TIME', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_TIME,
        payload: new Date('04/04/2020')
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_TAGS', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_TAGS,
        payload: 'edited tag'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('EDIT_POST_CONTENT', () => {
    it('should work corectly', function() {
      const action = {
        type: EDIT_POST_CONTENT,
        payload: 'edited content'
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })

  describe('INIT_POST', () => {
    it('should work corectly', function() {
      jest.spyOn(Date, 'now').mockReturnValue(new Date('04/04/2020'))
      const action = {
        type: INIT_POST
      }
      expect(postReducer({}, action)).toMatchSnapshot()
    })
  })
})
