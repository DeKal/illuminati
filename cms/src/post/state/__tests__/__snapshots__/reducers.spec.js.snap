// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Reducer CREATE_NEW_POST toSuccess should work corectly 1`] = `
Object {
  "content": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Array [
      "test",
      "test1",
    ],
    "time": "test",
    "title": "test",
  },
  "editedContent": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Array [
      "test",
      "test1",
    ],
    "time": "test",
    "title": "test",
  },
}
`;

exports[`Post Reducer EDIT_POST_AUTHOR should work corectly 1`] = `
Object {
  "editedContent": Object {
    "author": "edited author",
  },
}
`;

exports[`Post Reducer EDIT_POST_BRIEF should work corectly 1`] = `
Object {
  "editedContent": Object {
    "brief": "edited brief",
  },
}
`;

exports[`Post Reducer EDIT_POST_CONTENT should work corectly 1`] = `
Object {
  "editedContent": Object {
    "content": "edited content",
  },
}
`;

exports[`Post Reducer EDIT_POST_SLUG should work corectly 1`] = `
Object {
  "editedContent": Object {
    "slug": "edited-slug",
  },
}
`;

exports[`Post Reducer EDIT_POST_TAGS should work corectly 1`] = `
Object {
  "editedContent": Object {
    "tag": "edited tag",
  },
}
`;

exports[`Post Reducer EDIT_POST_TIME should work corectly 1`] = `
Object {
  "editedContent": Object {
    "time": "04/04/2020",
  },
}
`;

exports[`Post Reducer EDIT_POST_TITLE should work corectly 1`] = `
Object {
  "editedContent": Object {
    "title": "edited title",
  },
}
`;

exports[`Post Reducer FETCH_POST toRequest should work corectly 1`] = `
Object {
  "loading": true,
}
`;

exports[`Post Reducer FETCH_POST toSuccess should work corectly 1`] = `
Object {
  "content": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Object {
      "test": true,
      "test1": true,
    },
    "time": "test",
    "title": "test",
  },
  "editedContent": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Object {
      "test": true,
      "test1": true,
    },
    "time": "test",
    "title": "test",
  },
  "loading": false,
}
`;

exports[`Post Reducer INIT_POST should work corectly 1`] = `
Object {
  "content": Object {
    "author": "",
    "brief": "",
    "commentNum": 0,
    "content": "",
    "id": "",
    "likeNum": 0,
    "slug": "",
    "tag": Object {},
    "time": 2020-04-04T12:00:00.000Z,
    "title": "",
    "viewNum": 0,
  },
  "editedContent": Object {
    "author": "",
    "brief": "",
    "commentNum": 0,
    "content": "",
    "id": "",
    "likeNum": 0,
    "slug": "",
    "tag": Object {},
    "time": 2020-04-04T12:00:00.000Z,
    "title": "",
    "viewNum": 0,
  },
}
`;

exports[`Post Reducer SAVE_EDITED_CONTENT toSuccess should work corectly 1`] = `
Object {
  "content": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Array [
      "test",
      "test1",
    ],
    "time": "test",
    "title": "test",
  },
  "editedContent": Object {
    "author": "test",
    "brief": "test",
    "content": "test",
    "id": "test",
    "slug": "test",
    "tag": Array [
      "test",
      "test1",
    ],
    "time": "test",
    "title": "test",
  },
}
`;

exports[`Post Reducer default should work corectly 1`] = `Object {}`;
